### PR TITLE
Video Overlay & Video Poster Button

### PR DESCRIPTION
**Pull request by harterc1**
Mockup: [https://lonelyplanet.slack.com/files/brad.potterbaum/F2XA3TP70/destination_video_placement.png](url)

- masthead - Removed play button and any ties to video_player
- video_player - Video playback logic - no visual piece to this anymore;  brightcove.js still inherits from this.
- video_poster_button - Fetches a video, if a video is playable, displays a "button" that contains the poster image of the video and the name of the video as configured in Brightcove.  It uses video_overlay to show an overlay upon clicking the "button".
- video_overlay - inherits from overlay.js; contains a brightcove embed code expected to be on the page already.

### CODE REVIEW

File: src/components/video_overlay/video_overlay_component.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import Overlay from "../overlay";
+import waitForTransition from "../../core/utils/waitForTransition";
+import Video from "../video";
+
+/**
+ * Video Overlay Component
+*/
+export default class VideoOverlay extends Overlay {
+
+    initialize (options) {
```
Comment (jcreamer898) [inline src/components/video_overlay/video_overlay_component.js]: A bit nitpicky, but just noticed 4 space tabs? Should be 2 me thinks...

File: src/components/video_overlay/video_overlay_component.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        }
+        let width = maxWidth;
+        let height = width / ratio;
+
+        if (height > maxHeight) {
+            height = maxHeight;
+            width = height * ratio;
+        }
+
+        $(this.player.videoEl).css({width: width, height: height});
```
Comment (jcreamer898) [inline src/components/video_overlay/video_overlay_component.js]: `{ width: width, height: height }`

File: src/components/video_overlay/video_overlay_component.js (hunk 1 above)
Comment (harterc1) [inline src/components/video_overlay/video_overlay_component.js]: np

File: src/components/video_overlay/video_overlay_component.js (hunk 2 above)
Comment (harterc1) [inline src/components/video_overlay/video_overlay_component.js]: Thanks for the review.  This is just a comment regarding spacing right?

**Pull request merged**